hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -0,0 +1,68 @@
+# *****************************************************************************
+# Copyright (c) 2019, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import operator
+
+from numba import types
+from numba.extending import (
+    types,
+    overload,
+    overload_method
+)
+
+
+from hpat.hiframes.pd_series_ext import SeriesType as HPATPandasSeriesType
+
+
+'''
+Pandas Series (https://pandas.pydata.org/pandas-docs/stable/reference/series.html)
+functions and operators definition in HPAT
+Also, it contains Numba internal operators which are required for Series type handling
+'''
+
+
+@overload(operator.getitem)
+def hpat_pandas_series_getitem(self, idx):
+    '''
+    Internal Numba operator getitem implementation
+    Algorithm: result = series[idx]
+    Where:
+        series: pandas.series
+        idx: integer number
+    Test:  python -m hpat.runtests hpat.tests.test_series.TestSeries.test_static_getitem_series1
+    '''
+
+    if not isinstance(self, HPATPandasSeriesType):
+        raise numba_TypingError('The object must be a pandas.series')
+
```

Comment:
In other places overloads return nothing when meeting types that they are not handling (like here: https://github.com/numba/numba/blob/60d2bddd0a9ddf84cf900a70e4b7e17bcbe5dc87/numba/unicode.py#L492). I think we should follow this practice and not throw exceptions. Numba will then take care of error reporting (exceptions thrown here will be lost anyway).

Reply:
https://github.com/numba/numba/blob/60d2bddd0a9ddf84cf900a70e4b7e17bcbe5dc87/numba/unicode.py#L694

Reply:
ljust is not a standard operator!